library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java

```diff
@@ -881,6 +902,18 @@ public SkuDetails getSubscriptionListingDetails(String productId)
 		return getSkuDetails(productIdList, Constants.PRODUCT_TYPE_SUBSCRIPTION);
 	}
 
+    @NonNull
+    public List<SkuDetails> getPurchaseListingDetailsOrThrow(@NonNull ArrayList<String> productIdList) throws Exception
+	{
+		return getSkuDetailsOrThrow(productIdList, Constants.PRODUCT_TYPE_MANAGED);
+	}
+
+    @NonNull
+    public List<SkuDetails> getSubscriptionListingDetailsOrThrow(@NonNull ArrayList<String> productIdList) throws Exception
```

Comment:
Last nitpick: @dread-uo if we can add a short JavaDoc description for these 2 new methods, that would be ❤️